configs/usercall.toml

```diff
@@ -0,0 +1,34 @@
+# This example configuration covers detailed configuration for the usercall sampler
+
+[general]
+listen = "0.0.0.0:4242"
+
+[samplers.usercall]
+enabled = true
+
+
+# The quickest and easiest way to configure a library to be configured is to use the lib_search
+# configuration. This will look through some common library paths for the specified library and
+# add probes for the specified functions.
+[samplers.usercall.lib_search]
+# search for libcurl, and probe the function curl_global_init
+curl = ["curl_global_init"]
+
+# If the default search behavior doesn't work for you, you can specify the exact path to the library
+# you would like to probe by using the lib_files configuration option. If both a lib_search and
+# lib_file are specified, the lib_file will be used. Below is an example of sampling two different
+# versions of libkrb.
+#
+# WARNING: You cannot attach multiple probes to the same function.
+[samplers.usercall.lib_files.krb3]
```

Comment:
For the config - I'm wondering if it might make sense to do:
```
[samplers.usercall]
enabled = true

[[samplers.usercall.libraries]]
name = "foo"
paths = [ "/path/to/libfoo.so" ]
functions = [ "bar", "baz" ]

[[samplers.usercall.libraries]]
name = "qux"
functions = ["qux_init" ]

[[samplers.usercall.libraries]]
name = "grault5"
paths = [ "/path/to/libgrault.so.5" ]
functions = [ "garply" ]

[[samplers.usercall.libraries]]
name = "grault3"
paths = [ "/path/to/libgrault.so.5" ]
functions = [ "garply" ]

[[samplers.usercall.libraries]]
name = "grault"
paths = [ "/path/to/libgrault.so.5", ""/path/to/libgrault.so.3" ]
functions = [ "garply" ]
```

This might be a little more verbose in the config, but it might offer us better flexibility if we want to add additional keys in the future - currently we're treating the keys as paths, which may limit what we can do in the future without a breaking change.

I also think you could use this to differentiate between what's currently 'lib_files' vs 'lib_search' by conditioning on if there are paths specified within the library table.

Reply:
The only issue I see with this configuration is that we may have different functions in different versions of the libs. 

We could do it like you suggested above and just drop the vector of paths. We lose the concept of 'combined' probes, but probably we can do that combination during querying. I guess there are pluses and minuses to both. I'm leaning slightly towards what you suggested with a single path and just getting rid of combined probes. Happy to write up whichever you prefer.

Reply:
Okay, got some updates in that should get the config to be of the format we discussed.